openpype/pipeline/plugin_discover.py

```diff
@@ -0,0 +1,264 @@
+                and not added to result.
+
+        Returns:
+            DiscoverResult: Object holding succesfully discovered plugins,
+                ignored plugins, plugins with missing abstract implementation
+                and duplicated plugin.
+        """
+
+        if not ignore_classes:
+            ignore_classes = []
+
+        result = DiscoverResult()
+        plugin_names = set()
+        registered_classes = self._registered_plugins.get(superclass) or []
+        registered_paths = self._registered_plugin_paths.get(superclass) or []
+        for cls in registered_classes:
+            if cls is superclass or cls in ignore_classes:
+                result.ignored_plugins.add(cls)
+                continue
+
+            if inspect.isabstract(cls):
+                result.abstract_plugins.append(cls)
+                continue
+
+            class_name = cls.__name__
+            if class_name in plugin_names:
+                result.duplicated_plugins.append(cls)
+                continue
+            plugin_names.add(class_name)
+            result.plugins.append(cls)
+
+        # Include plug-ins from registered paths
+        for path in registered_paths:
+            modules, crashed = modules_from_path(path)
+            for item in crashed:
+                filepath, exc_info = item
+                result.crashed_file_paths[filepath] = exc_info
+
+            for item in modules:
+                filepath, module = item
+                for cls in classes_from_module(superclass, module):
+                    if cls is superclass or cls in ignore_classes:
+                        result.ignored_plugins.add(cls)
+                        continue
+
+                    if inspect.isabstract(cls):
+                        result.abstract_plugins.append(cls)
+                        continue
+
+                    if not allow_duplicates:
+                        class_name = cls.__name__
+                        if class_name in plugin_names:
+                            result.duplicated_plugins.append(cls)
+                            continue
+                        plugin_names.add(class_name)
+
+                    result.plugins.append(cls)
+
+        self._last_discovered_plugins[superclass] = list(
+            result.plugins
+        )
+        result.print_report()
+        return result
+
+    def register_plugin(self, superclass, cls):
+        """Register an individual `obj` of type `superclass`
+
+        Arguments:
+            superclass (type): Superclass of plug-in
+            cls (object): Subclass of `superclass`
+        """
+
+        if superclass not in self._registered_plugins:
+            self._registered_plugins[superclass] = list()
+
+        if cls not in self._registered_plugins[superclass]:
+            self._registered_plugins[superclass].append(cls)
+
+    def register_plugin_path(self, superclass, path):
+        """Register a directory of one or more plug-ins
```

Comment:
This currently reads to me as if it could be for one or more plug-in types.
```suggestion
        """Register a directory containing plug-ins of type `superclass`
```